WordPress/wordpress-playground
# PHP : Give access to command arguments if array type is given in php ^7.4 proc_open function

## What is this PR doing?

Based on the issue #930 

For now, if you have this PHP code : 

```
$process = proc_open( [ 'ls', '-a' ], $descriptorspec, $pipes );
```

the `php.setSpawnHandler( command => ... )` `command` parameter will only return the first element of the array.

## What problem is it solving?

It gives the ability to get the overall command in one string separated by spaces. In the example above `ls -a`.

## Testing Instructions

`proc.php`

```
$process = proc_open( [ 'ls', '-a' ], [ 0 => ['pipe', 'r'], 1 => ['pipe', 'w'], 2 => ['pipe', 'w'] ], $pipes );

sleep( 1 );
```

`cli.js`

```
import { NodePHP } from '@php-wasm/node';


const php = await NodePHP.load( '8.2' );

php.setSpawnHandler( command => console.log( command ) );

php.cli( [ 'php', ...process.argv.slice( 2 ) ] ).finally( () => process.exit( 0 ) );
```

`node wasm/cli.js proc.php`

returns :

```
ls -a

Node.js v18.18.2
```

@adamziel 

P.S.: It seems `argv` is no longer needed. It was used with `command` and `execvp` to probably launch the command with its arguments. I decided to clean the file from the unused variable.

P.S.2 : It seems there is no more sign of `PHP_WIN32` in `proc_open7.4.c` file but there are still in `proc_open7.0`. Should I make another PR to clean these files from these unneeded definitions, for clarity ?

P.S.3 : I only modified the `proc_open7.4.c` file for now, but I probably have to run some scripts, what are the steps ?
